jax/_src/numpy/lax_numpy.py

```diff
@@ -3157,8 +3157,24 @@ def _supports_buffer_protocol(obj):
 
 deprecations.register("jax-numpy-array-none")
 
-@util.implements(np.array, lax_description=_ARRAY_DOC)
+@util.implements(np.array, lax_description=_ARRAY_DOC, extra_params="""
+device : :py:class:`Device`, :py:class:`Sharding`, optional
+  The (optional) :py:class:`Device`, :py:class:`Sharding`,
+  representing the device(s) to which created array should be
+  transferred. If given, then the result is committed to the device(s).
+""")
 def array(object: Any, dtype: DTypeLike | None = None, copy: bool = True,
+          order: str | None = "K", ndmin: int = 0,
+          *, device: xc.Device | Sharding | None = None) -> Array:
+  # TODO(vfdev-5): optimize putting the array directly on the device specified
+  # instead of putting it on default device and then on the specific device
+  output = _array(object, dtype=dtype, copy=copy, order=order, ndmin=ndmin)
```

Comment:
`array` is central enough that I don't think we should punt on this. Let's handle `device` directly within the existing implementation.

Reply:
@jakevdp I tried to insert device_put into array implementation and removed `_array` function. Let me know if we can do better.

> That will also let us use jnp.asarray(..., device=device) in all the other functions where we're adding a device argument, and we won't have to add the same boilerplate to every function definition.

If I understand your point here correctly, we can add device arg to `jnp.asarray` as well in this PR ?

Reply:
Right, `asarray` is just a thin wrapper around `array`, so once `array` is implemented `asarray` is trivial.